staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -226,9 +226,10 @@ func isCancelError(err error) bool {
 
 func (wc *watchChan) run(initialEventsEndBookmarkRequired, forceInitialEvents bool) {
 	watchClosedCh := make(chan struct{})
-	go wc.startWatching(watchClosedCh, initialEventsEndBookmarkRequired, forceInitialEvents)
-
 	var resultChanWG sync.WaitGroup
+
+	resultChanWG.Add(1)
+	go wc.startWatching(watchClosedCh, initialEventsEndBookmarkRequired, forceInitialEvents, &resultChanWG)
```

Comment:
This is an unrelated fix that I found - we need to ensure that we won't close resultChan before startWatching is finished too.

Reply:
separating the Add and Done makes this tricky to follow... can we do this like this intead and avoid plumbing resultChanWG into startWatching?

```go
resultChanWG.Add(1)
go func(){
  defer wg.Done()
  wc.startWatching(watchClosedCh, initialEventsEndBookmarkRequired, forceInitialEvents)
}()
```

Reply:
(also, this is a new issue introduced when https://github.com/kubernetes/kubernetes/pull/131162 made startWatching send errors directly to resultCh, right? just making sure we don't need to separate this out and pick it to older minors)

Reply:
Updated

Yes - that's a new thing introduced in #131162